File: temas/delegacion/instrumento.cpp
#include "iostream"
using namespace std;

class Instrumento{
    public:
        virtual void afinar() = 0;
        virtual void tocar() = 0;
        const string getNombreInstrumento() { return nombreInstrumento; }
    protected:
        string nombreInstrumento;
};

class Viento: public Instrumento{
    public:
        Viento(string name){nombreInstrumento = name;}
        void afinar(){ cout << "Afinando instrumento de Viento: " << getNombreInstrumento() << endl;}
        void tocar() { cout << "Tocando instrumento de Viento: " << getNombreInstrumento() << endl;}

};

class Cuerda: public Instrumento{
    public:
        Cuerda(string name){nombreInstrumento = name;}
        void afinar(){ cout << "Afinando instrumento de Cuerda: " << getNombreInstrumento() << endl;}
        void tocar(){ cout << "Tocando instrumento de Cuerda: " << getNombreInstrumento() << endl;}
};

class Percusion: public Instrumento{
    public:
        Percusion(string name){nombreInstrumento = name;}
        void afinar(){ cout << "Afinando instrumento de Percusión: " << getNombreInstrumento() << endl;}
        void tocar(){ cout << "Tocando instrumento de Percusión: " << getNombreInstrumento() << endl;}
};

File: temas/delegacion/python/instrumento.py

class Instrumento:
	def __init__(self,name):
		self._nombreInstrumento = name
	def getNombreInstrumento(self):
		return self._nombreInstrumento

class Viento(Instrumento):
	def __init__(self,name):
		self._nombreInstrumento = name
	def afinar(self):
		print "Afinando instrumento de Viento: ", self.getNombreInstrumento()
	def tocar(self):
		print "Tocando instrumento de Viento: ", self.getNombreInstrumento()

class Cuerda(Instrumento):
	def __init__(self,name):
		self._nombreInstrumento = name
	def afinar(self):
		print "Afinando instrumento de Cuerda: ", self.getNombreInstrumento()
	def tocar(self):
		print "Tocando instrumento de Cuerda: ", self.getNombreInstrumento()


class Percusion(Instrumento):
	def __init__(self,name):
		self._nombreInstrumento = name
	def afinar(self):
		print "Afinando instrumento de Percusion: ", self.getNombreInstrumento()
	def tocar(self):
		print "Tocando instrumento de Percusion: ", self.getNombreInstrumento()